api.go

```diff
@@ -983,6 +984,60 @@ func (api *API) Info() serverInfo {
 	}
 }
 
+// EnsureOperatingNode determine if the current node is the destination for execution if not forward on to correct node
+func (api *API) EnsureOperatingNode(indexName string, w http.ResponseWriter, r *http.Request) bool {
+
+	values := r.URL.Query()
+
+	if _, keyImport := values["keyImport"]; keyImport {
+		// if not coordinator..forward on
+		if !api.Node().IsCoordinator {
+			api.forwardTo(w, r, api.getCoordinator())
+			return true
+		}
+		return false
+	}
+
+	if shardS, hasShard := values["shard"]; hasShard {
+		shard, err := strconv.ParseUint(shardS[0], 10, 64)
```

Comment:
I take it shardS is a []string that's guaranteed to always be of length 1?

Reply:
is it possible that it might exist, but have length 0?

Reply:
just an artifact of using the  go stdlib URL.Query api.  The only correct way to call this method is with shard=SHARDNUM and anything other than that would be an error.  So just error handling needs to be tightened up